src/frontends/lean/parser.cpp

```diff
@@ -533,6 +533,10 @@ void parser::add_local_level(name const & n, level const & l, bool is_variable)
 }
 
 void parser::add_local_expr(name const & n, expr const & p, bool is_variable) {
+    if (!m_in_quote) {
+        // certainly not in a pattern
+        m_in_pattern = false;
```

Comment:
Ah well, it does break the error message in https://github.com/leanprover/lean/blob/master/tests/lean/inaccessible2.lean#L11. Suggestions?